src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -1307,6 +1307,16 @@ def build_vmss_resource(cmd, name, computer_name_prefix, location, tags, overpro
 
         if not rolling_upgrade_policy:
             del rolling_upgrade_policy
+    if upgrade_policy_mode and cmd.supported_api_version(min_api='2018-10-01',
+                                                         operation_group='virtual_machine_scale_sets'):
+        vmss_properties['upgradePolicy']['automaticOSUpgradePolicy'] = {}
+        automatic_os_upgrade_policy = vmss_properties['upgradePolicy']['automaticOSUpgradePolicy']
+
+        if enable_auto_os_upgrade is not None:
+            automatic_os_upgrade_policy['enableAutomaticOSUpgrade'] = enable_auto_os_upgrade
+
+        if not automatic_os_upgrade_policy:
+            del automatic_os_upgrade_policy
```

Comment:
Why is this logic needed?

Reply:
I emulated the logic of the `rolling_upgrade_policy` part, which is a concept similar to `automatic_os_upgrade_policy`.

Reply:
OK, I guess its purpose is to not pass it to the server when `rolling_upgrade_policy` is not available